### ISSUE TITLE
Expose all auth-provider config keys

### ISSUE THREAD

**Issue by tewe** [kubernetes/kubectl#1093]
`kubectl config unset users.foo.auth-provider.azure.refresh-token` dies with "error: unable to parse one or more field values" because `pkg/cmd/config/navigation_step_parser.go` cannot deal with pointers and `*AuthProviderConfig` is one.

**What would you like to be added**:

Allow `kubectl config` to un/set any field in the YAML, particularly `auth-provider` settings.

**Why is this needed**:

When giving people access to a cluster it is often necessary to deal with authentication issues, and sending them command lines is easier for everybody than instructions on how to edit the kubeconfig by hand.

**Comment by tewe** [comment]
#396

**Comment by eddiezane** [comment]
/triage accepted
/help

We need to take a look if these types are pulled in by config as a start.

**Comment by calvin0327** [comment]
@eddiezane I tried to reslove this issue, the `AuthProviderConfig` is a `Ptr` type of struct that define field of `Name` and `Config`.  there is not a `case` to corresponding to the type in the switch.  can you give me some advice?

**Comment by mpuckett159** [comment]
After some review, this appears (I think) to be an issue with the client-go API rather than kubectl itself. The way that the paths are walked from the config key assumes that the AuthProvider field will be a `map[string]*AuthProviderConfig`. This is how we get from `users.foo` to `auth-provider` even though there is no valid field name in `AuthInfo` named `foo`.

Unsure if there should be a workaround for this specific struct in kubectl or if we should be open an issue with client-go repo to update the `Config.AuthInfo.AuthProvider` field to be a map instead of a pointer.

**Comment by mpuckett159** [comment]
Also of note it looks like `ExecConfig` (e.g. `users.foo.exec.command` and similar) suffers the same issue.

**Comment by mpuckett159** [comment]
@tewe Does `kubectl config set-credential` solve your use case? I think this may be a bit of a confusion in the docs (and maybe just update the error messages to point to the proper commands) here rather than a feature request.

**Comment by tewe** [comment]
(My solution has been [kubelogin](https://github.com/Azure/kubelogin))

Can `--auth-provider-arg` *remove* a key? Then it should be equivalent.

(In the case of the refresh token, setting it to a bogus value probably has the same effect)

**Comment by mpuckett159** [comment]
It looks like if you were to use `kubectl config set-credentials test --auth-provider=test --auth-provider-arg=refresh-token-` (notice the `-` at the end) it will remove the specified arg key.

**Comment by tewe** [comment]
Thanks! I'll leave this open due to the confusing error message.

**Comment by gswallow** [comment]
Is this still broken?  The last pull request seems to have been closed.

**Comment by mpuckett159** [comment]
Yes, unfortunately this became a rather monumental task. I had something working but it was decided that the sig did not wish to extend the existing dot delimited syntax (which makes sense) and instead we would look to moving to using jsonpath instead. Unfortunately I have not had time to do this and I don't think anyone else has tried to pick it up.
